baseband_tasks/dispersion.py

```diff
@@ -170,7 +173,117 @@ class Dedisperse(Disperse):
     See Also
     --------
     baseband_tasks.fourier.fft_maker : to select the FFT package used.
+    baseband_tasks.dispersion.Disperse : for coherent dispersion
+    baseband_tasks.dispersion.DedisperseSamples : for incoherent dedispersion
+    """
+
+    def __init__(self, ih, dm, reference_frequency=None,
+                 samples_per_frame=None, frequency=None, sideband=None):
+        super().__init__(ih, -dm, reference_frequency=reference_frequency,
+                         samples_per_frame=samples_per_frame,
+                         frequency=frequency, sideband=sideband)
+
+    @property
+    def dm(self):
+        return -self._dm
+
+
+class DisperseSamples(ShiftSamples):
+    """Incoherently shift a time stream to give it a dispersive time delay.
+
+    This task does not handle any in-channel dispersive smearing, but only
+    shifts the samples according to the mid-channel frequency.
 
+    Parameters
+    ----------
+    ih : task or `baseband` stream reader
+        Input data stream, with time as the first axis.
+    dm : float or `~baseband_tasks.dm.DispersionMeasure` quantity
+        Dispersion measure to disperse with.  If negative, will dedisperse,
+        but clearer to use `~baseband_tasks.dispersion.DedisperseSamples`.
+    reference_frequency : `~astropy.units.Quantity`
+        Frequency to which the data should be dispersed.  Can be an array.
+        By default, the mean frequency.
+    samples_per_frame : int, optional
+        Number of dispersed samples which should be produced in one go.
+        The number of input samples used will be larger to avoid wrapping.
+        If not given, as produced by the minimum power of 2 of input
+        samples that yields at least 75% efficiency.
+    frequency : `~astropy.units.Quantity`, optional
+        Frequencies for each channel in ``ih``.
+        Default: taken from ``ih`` (if available).
+    sideband : array, optional
+        Whether frequencies in ``ih`` are upper (+1) or lower (-1) sideband.
+        Default: taken from ``ih`` (if available). Note that while this is
+        only used if the data is real (to calculate the mid-channel
+        frequency), it should always be passed in together with ``frequency``,
+        since otherwise other tasks cannot interpret frequency correctly.
+
+    See Also
+    --------
+    baseband_tasks.dispersion.DedisperseSamples : for incoherent dedispersion
+    baseband_tasks.dispersion.Disperse : for coherent dispersion
+    """
+
+    def __init__(self, ih, dm, reference_frequency=None,
+                 samples_per_frame=None, frequency=None, sideband=None):
+        # Compute the time shift
+        dm = DispersionMeasure(dm)
+
+        # Set possible missing frequency/sideband attributes
+        if frequency is not None or sideband is not None:
+            ih = SetAttribute(ih, frequency=frequency, sideband=sideband)
```

Comment:
Do we want to overwrite the upstream frequency if it exists? Should we give a warning?

Reply:
The other tasks also work like this, that if you pass in `frequency`, the underlying values get ignored. I think it is OK to stick with that.